content/5_Plat/RURQ.mdx

```diff
@@ -221,9 +221,171 @@ queries.
 	</Resource>
 </Resources>
 
-### Implementation
+### Solution - Range Updates & Sums
 
-<IncompleteSection />
+<Spoiler title="Solution">
+
+This question asks you to support the following types of queries. 
+
+- Add a value to all elements within the range $[a,b]$
+
+- Set all values within the range $[a,b]$ to a certain value 
+
+- Find the sum of all values in range $[a,b]$
+
+Consider the first two types of queries, a lazy tag will be created in each node of the tree for each type. In this solution, $lz1$ will represent the lazy tag for the range add query and $lz2$ will represent the lazy tag for the range set query. 
+
+Given the two different update queries, a total of four different situations might take place as a result of any update. 
+
+- Range add ($lz1$) when $lz2$ equals 0: Simply add the new value to the pre-existing value 
+
+- Range add ($lz1$) when $lz2$ doesnt equal to 0: Add the new value to $lz2$ and clear $lz1$ 
+
+- Range set when $lz1$ equals 0: Simply update the $lz2$ value 
+
+-Range set when $lz1$ doesn't equal to 0: Again simply update the $lz2$ value since a set update will override all previous add updates 
```

Comment:
> -Range

Did you check this in liveupdate?

Reply:
Yes I have checked this in liveupdate. The unordered list function works fine on my side.

Reply:
Typo on my part. Addressed.